explorer/interpreter/resolve_names.cpp

```diff
@@ -856,7 +856,13 @@ auto NameResolver::ResolveNames(Declaration& declaration,
       CARBON_ASSIGN_OR_RETURN(auto target,
                               ResolveNames(alias.target(), *scope));
       if (target && isa<Declaration>(target->base())) {
-        alias.set_resolved_declaration(&cast<Declaration>(target->base()));
+        if (alias.resolved_declaration()) {
+          // Skip if the declaration is already resolved in a previous name resolution
+          // phase.
+          CARBON_CHECK(*alias.resolved_declaration() == target->base());
```

Comment:
```suggestion
          CARBON_CHECK(*alias.resolved_declaration() == &target->base());
```

I'm sure you'll figure this out on your own, but I think this fixes the build error. :)

(and maybe the suggestion will be easier to apply)